db/migrate/20240801034710_set_connected_apps_enabled_if_any.rb

```diff
@@ -0,0 +1,14 @@
+class SetConnectedAppsEnabledIfAny < ActiveRecord::Migration[7.0]
+  def up
+    enabled = []
+    enabled << "discover_regen" if ConnectedApp.discover_regen.any?
+    enabled << "affiliate_sales_data" if ConnectedApp.affiliate_sales_data.any?
+
+    Spree::Config.connected_apps_enabled = enabled.presence&.join(",")
+  end
+end
+
+class ConnectedApp < ApplicationRecord
+  scope :discover_regen, -> { where(type: "ConnectedApp") }
+  scope :affiliate_sales_data, -> { where(type: "ConnectedApps::AffiliateSalesData") }
+end
```

Comment:
We usually define these as sub-classes of the migration. The table name still works and it's definitely another namespace to the original model.

Thankfully, good_migrations ensures that we don't load the actual model here.